### DISCUSSION TITLE
Distributed monitoring similar to checkmk?

### DISCUSSION THREAD

**Discussion by patmaddox** [netbox-community/netbox#11177] [Q&A]
Hi there, I am evaluating monitoring systems and came across netbox.

One important feature I’m looking for is the ability to set up an instance in each of our clients’ cloud accounts, so they have access but only see the data for that account. Then we have a central instance that can view everything.

[checkmk calls this distributed monitoring](https://docs.checkmk.com/latest/en/distributed_monitoring.html) and I wonder if netbox can do something similar?

**Comment by mtinberg** [comment]
Netbox is not a monitoring system, is a database for storing IP/VLAN/Device/Interface inventory information to provide better documentation than spreadsheets and an API accessible database for config management/audit tools to use to create intended configs to deploy on equipment.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Pat Maddox ***@***.***>
Sent: Wednesday, December 14, 2022 6:26 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Distributed monitoring similar to checkmk? (Discussion #11177)


Hi there, I am evaluating monitoring systems and came across netbox.

One important feature I’m looking for is the ability to set up an instance in each of our clients’ cloud accounts, so they have access but only see the data for that account. Then we have a central instance that can view everything.

checkmk calls this distributed monitoring<https://docs.checkmk.com/latest/en/distributed_monitoring.html> and I wonder if netbox can do something similar?

**Comment by mtinberg** [comment]
Hit send too early 😉

That said Netbox does have a permissions system so you can define Tenants and Groups and restrict access to DB records so people can view/edit records they should be allowed to and not records they are not allowed to.  Take a look at https://demo.netbox.dev/static/docs/administration/permissions/ to show how to configure Django query filters so that a particular User/Group permission applies to a specific subset of objects (aka records/rows) in the DB model.  You can use these tools to achieve some level of multi-tenancy on the same Netbox instance.

You could also create individual instances for each customer, and possibly write sync tools to push/pull records from a central instance, but that has a lot more overhead and moving parts.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Pat Maddox ***@***.***>
Sent: Wednesday, December 14, 2022 6:26 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Distributed monitoring similar to checkmk? (Discussion #11177)


Hi there, I am evaluating monitoring systems and came across netbox.

One important feature I’m looking for is the ability to set up an instance in each of our clients’ cloud accounts, so they have access but only see the data for that account. Then we have a central instance that can view everything.

checkmk calls this distributed monitoring<https://docs.checkmk.com/latest/en/distributed_monitoring.html> and I wonder if netbox can do something similar?
